📚 DOCS: announce post (#91)
executablebooks/meta: docs/updates/2020-08-07-announce-book.md

# 2020-08-07 - Announcing the new Jupyter Book

*Note: this announcement is cross-posted between the [Jupyter Blog](https://blog.jupyter.org) and the [Executable Book Project updates blog](https://executablebooks.org/en/latest/updates/index.html)*

> Jupyter Book is an open source project for building beautiful, publication-quality books and documents from computational material. With this post, we're happy to announce that Jupyter Book has been re-written from the ground up, making it easier to install, faster to use, and able to create more complex publishing content in your books. It is now supported by [the Executable Book Project](https://executablebooks.org/en/latest/), an open community that builds open source tools interactive and executable documents in the Jupyter ecosystem and beyond.

Reviewer: "from computational material" sounds a bit odd to me, what does it mean?
(note: in the "What next?" section you use "computational content")

Maybe something like:

Jupyter Book is an open source project for combining text and data into beautiful, publication-quality books and documents.

Reviewer: "builds open source tools interactive" is there some words missing after tools? "builds open source tools for interactive"?

Author: I updated this so it includes "source material with computational content" per @jstac's suggestion above

> Jupyter Book is an open source project for building beautiful, publication-quality books and documents from computational material. With this post, we're happy to announce that Jupyter Book has been re-written from the ground up, making it easier to install, faster to use, and able to create more complex publishing content in your books. It is now supported by [the Executable Book Project](https://executablebooks.org/en/latest/), an open community that builds open source tools interactive and executable documents in the Jupyter ecosystem and beyond.

Reviewer 2: change "books and documents" to "books, websites and other documents"?

> Jupyter Book is an open source project for building beautiful, publication-quality books and documents from computational material. With this post, we're happy to announce that Jupyter Book has been re-written from the ground up, making it easier to install, faster to use, and able to create more complex publishing content in your books. It is now supported by [the Executable Book Project](https://executablebooks.org/en/latest/), an open community that builds open source tools interactive and executable documents in the Jupyter ecosystem and beyond.

Reviewer 2: change "computational material" to "source material including computational content"?

```{figure} images/jb-ui.png

The new Jupyter Book interface, highlighting how you can [insert code outputs into your content](https://jupyterbook.org/content/glue.html)
```

This post is a short overview of the new features in Jupyter Book, and gives some background on the future of the project.

You can also browse the [new Jupyter Book documentation](https://jupyterbook.org), or [check out the Jupyter Book GitHub repository](https://github.com/executablebooks/jupyter-book).

> **Note**: Jupyter Book is still in beta, so things are always changing. We'd love your feedback, ideas, and PRs for how to make the project better!

Reviewer: ````suggestion
```{note}
Jupyter Book is still in beta, so things are always changing. We'd love your feedback, ideas, and PRs for how to make the project better!
```
````

## What does the new Jupyter Book do?

The new version of Jupyter Book will feel very similar. However, it has a lot of new features due to the new Jupyter Book stack underneath (more on that later).

The new Jupyter Book has the following main features (with links to the relevant documentation for each):

> * **[Write publication-quality content in markdown](https://jupyterbook.org/content-types/markdown)**. You can write in either Jupyter markdown, or in [MyST Markdown, an extended flavor of markdown for scientific publishing](https://jupyterbook.org/content/myst).  This includes support for rich syntax such as [citations and cross-references](https://jupyterbook.org/content/citations), [math and equations](https://jupyterbook.org/content/math), and [figures](https://jupyterbook.org/content/figures).

Reviewer: nitpick, but I don't feel that the main features (i.e. the text in bold) stands out very well from the rest of the text, especially due to all the other links, it all just kind of blurs into one. Perhaps have the subsequent text as sub-bullet points or something?

<img width="599" alt="image" src="https://user-images.githubusercontent.com/2997570/89692610-1abe5800-d904-11ea-999a-78f587312835.png">

Author: @chrisjsewell what do you think about the layout on the current jupyterbook.org home page? Is that better?

Reviewer: https://github.com/executablebooks/jupyter-book/pull/864

Author: I've updated this so that it mimics the jupyterbook.org front page

* **[Add interactivity to your book](https://jupyterbook.org/interactive/launchbuttons)**. You can [toggle visibility of cells](https://jupyterbook.org/interactive/hiding), [connect with an online service like Binder](https://jupyterbook.org/interactive/launchbuttons), and include [interactive outputs from Jupyter](https://jupyterbook.org/interactive/interactive).
* **[Generate a variety of outputs](https://jupyterbook.org/start/build)**, including single- and multi-page websites, as well [as PDF outputs](https://jupyterbook.org/advanced/pdf).
* **[Build your book with a simple CLI](https://jupyterbook.org/reference/cli)** to quickly generate your books with one command, like so: `jupyter-book build mybook/`

These are just a few of the major changes that we've made. For a more complete idea of what you can do, [check out the Jupyter Book documentation](https://jupyterbook.org)

## Major changes

In the next few sections we'll talk about some major changes that you may notice in Jupyter Book.

### An enhanced flavor of markdown

> The biggest enhancement to Jupyter Book is support for the [MyST Markdown language](https://myst-parser.readthedocs.io/en/latest/). MyST is a flavor of markdown that implements all of the features of Sphinx, allowing you to write scientific publications in markdown. It draws inspiration from RMarkdown and the reStructuredText ecosystem of tools. Anything you can do in Sphinx, you can do with MyST as well.

Reviewer: Include the acronym expansion somewhere? i.e. Markedly Structured Text (MyST)

Reviewer: link to sphinx docs and/or mention what it is (e.g. the sphinx static site generator)?

Reviewer: ah you do mention what sphinx is later on. Maybe better earlier but not sure

Author: added a bit more explanation of sphinx + a link, and added acronym expansion

A sample citation. Here we see how you can include citation syntax in-line with your markdown, and then insert a bibliography later on in your page.
```

And here's how you can include a figure:

```{figure} images/figure-example.png

A sample figure. Here we see how you can insert a figure with a caption into your book,
and control some aspects of how the figure is displayed with options.
```

### A smarter build system

While the old version of Jupyter Book used a combination of Python and Jekyll to build your book's HTML, the new Jupyter Book uses Python all the way through. This means that building the HTML for your book is as simple as:

```
jupyter-book build mybookname/
```

In addition, the new build system leverages [Jupyter Cache](https://jupyter-cache.readthedocs.io/) to execute notebook content *only if the code is updated*, and to insert the outputs from the cache at build time. This saves you time by avoiding the need to re-execute code that hasn't been changed.

```{figure} images/build-process.gif

An example build process. Here the `jupyter-book` command-line interface is used to convert a collection of content into an HTML book.
```

### More book output types

> By leveraging Sphinx, Jupyter Book will be able to support more complex outputs than *just* an HTML website. For example, we are currently prototyping **[PDF Outputs](https://jupyterbook.org/advanced/pdf.html)**, both via HTML as well as via Latex. This gives Jupyter Book more flexibility to generate the right book for your usecase.

Reviewer: ```suggestion
By leveraging Sphinx, Jupyter Book will be able to support more complex outputs than *just* an HTML website.
For example, we are currently prototyping **[PDF Outputs](https://jupyterbook.org/advanced/pdf.html)**, both via HTML as well as via LaTeX.
This gives Jupyter Book more flexibility to generate the right book for your use case.
```

You can also run Jupyter Book on **[individual pages](https://jupyterbook.org/start/build.html#build-a-standalone-page)**. This means that you can write single-page content (like a scientific article) entirely in Markdown.

### A new stack

The biggest change under-the-hood is that Jupyter Book now uses [the Sphinx documentation engine](https://www.sphinx-doc.org/en/master/) instead of Jekyll for building books. By leveraging the Sphinx ecosystem, Jupyter Book can more effectively build on top of community tools, and can contribute components back to the broader community.

> Instead of being a single repository, the  old Jupyter Book repository has now been separated into several modular tools. **Each of these tools can be used on their own in your Sphinx documentation**, and they can be coordinated together via Jupyter Book:

Reviewer: ```suggestion
Instead of being a single repository, the old Jupyter Book repository has now been separated into several modular tools.
**Each of these tools can be used on their own in your Sphinx documentation**, and they can be coordinated together *via* Jupyter Book:
```
